File: ITServiceDesk.Core/Entities/Attachment.cs
namespace ITServiceDesk.Core.Entities;

public class Attachment : BaseEntity
{
    public string FileName { get; set; } = string.Empty;
    public string FilePath { get; set; } = string.Empty;
    public string ContentType { get; set; } = string.Empty;
    public long FileSize { get; set; }
    
    // Navigation Properties
    public Guid? TicketId { get; set; }
    public Ticket? Ticket { get; set; }
    
    public Guid? CommentId { get; set; }
    public Comment? Comment { get; set; }
}

File: ITServiceDesk.Client/src/types/attachment.ts
export interface AttachmentResponseDto {
    id: string;
    fileName: string;
    filePath: string;
    contentType: string;
    fileSize: number;
    ticketId?: string;
    commentId?: string;
    uploaderId?: string;
    uploaderName?: string;
    uploaderRole?: string;
    createdAt: string;
}

export interface AttachmentCreateDto {
    file: File;
    ticketId?: string;
    commentId?: string;
}
